### PR TITLE
Change order of regularization in psislw (#1941)

### LINKED ISSUE #1941: Change order of regularization in psislw
The [PSIS paper](https://arxiv.org/abs/1507.02646) is ambiguous as to whether the regularization of the shape parameter of the generalized Pareto distribution (in appendix C) should be performed before or after the scale parameter is estimated. ArviZ currently does it before, while loo does it afterwards. On Slack, @avehtari said:

> loo package implementation is intentional, and based on my experiments produces better results than the alternative. The joint distribution has strong dependency, which is further exaggerated by the use of profile likelihood and point estimates. It seems it's better to let the prior affect only k.

So I believe all we need to do is swap these two lines:

https://github.com/arviz-devs/arviz/blob/045ddd3b7c3f7ff3ab082cfc0f207e09dce9e629/arviz/stats/stats.py#L955-L956

### PR DESCRIPTION
**Pull request by OrchiDorchi**
## Description
Resolves #1941. Changed the order of regularization of the shape parameter of the generalized Pareto distribution and estimation of scale parameter.

## Checklist

- [ ] Follows [official](https://github.com/arviz-devs/arviz/blob/main/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/main/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@
 
 ## v0.11.3 (2021 Oct 1)
 ### New features
+* Change order of regularization in psislw ([1943](https://github.com/arviz-devs/arviz/pull/1943))
```
Comment (sethaxen) [inline CHANGELOG.md:24]:
```suggestion
* Change order of regularization in `psislw` ([1943](https://github.com/arviz-devs/arviz/pull/1943))
```

**Review by sethaxen — COMMENTED** [review]
One small formatting tweak to the changelog, then this LGTM! Thanks for the fix!

**Comment by OrchiDorchi** [comment]
I just updated CHANGELOG but test that passed last week failed now. Is there anything I can do?

**Review by sethaxen — APPROVED** [review]

**Comment by OrchiDorchi** [comment]
Thanks for your time, I am waiting for merge.

**Pull request merged**